### ISSUE TITLE
Postgres dialect doesn't support UPDATE FROM

### ISSUE THREAD

**Issue by ralsina** [sqlfluff/sqlfluff#913]
When linting this SQL:

```sql
UPDATE foo SET quux=42 FROM bat WHERE bat.bar=foo.bar and bat.baz < foo.baz
```

## Expected Behaviour

It should parse, since it's valid syntax:

![image](https://user-images.githubusercontent.com/1579/113288067-d59ccb00-92c4-11eb-85a7-7c1716009c88.png)

## Observed Behaviour

Doesn't parse.

```
$ sqlfluff lint --dialect postgres x.sql
== [x.sql] FAIL
L:   1 | P:  22 |  PRS | Found unparsable section: ' FROM bat WHERE
                       | bat.bar=foo.bat and bat....'

$ sqlfluff parse --dialect postgres x.sql
[0](1, 1, 1)        |file:
[0](1, 1, 1)        |    statement:
[0](1, 1, 1)        |        update_statement:
[0](1, 1, 1)        |            keyword:                                          'UPDATE'
[6](1, 1, 7)        |            whitespace:                                       ' '
[7](1, 1, 8)        |            table_reference:
[7](1, 1, 8)        |                identifier:                                   'foo'
[10](1, 1, 11)      |            whitespace:                                       ' '
[11](1, 1, 12)      |            set_clause_list:
[11](1, 1, 12)      |                keyword:                                      'SET'
[14](1, 1, 15)      |                [META] indent:
[14](1, 1, 15)      |                whitespace:                                   ' '
[15](1, 1, 16)      |                set_clause:
[15](1, 1, 16)      |                    column_reference:
[15](1, 1, 16)      |                        identifier:                           'bar'
[18](1, 1, 19)      |                    comparison_operator:                      '='
[19](1, 1, 20)      |                    literal:                                  '42'
[21](1, 1, 22)      |                [META] dedent:
[21](1, 1, 22)      |            unparsable:                                       !! Expected: 'Nothing...'
[21](1, 1, 22)      |                whitespace:                                   ' '
[22](1, 1, 23)      |                raw:                                          'FROM'
[26](1, 1, 27)      |                whitespace:                                   ' '
[27](1, 1, 28)      |                raw:                                          'bat'
[30](1, 1, 31)      |                whitespace:                                   ' '
[31](1, 1, 32)      |                raw:                                          'WHERE'
[36](1, 1, 37)      |                whitespace:                                   ' '
[37](1, 1, 38)      |                raw:                                          'bat'
[40](1, 1, 41)      |                raw:                                          '.'
[41](1, 1, 42)      |                raw:                                          'bar'
[44](1, 1, 45)      |                raw:                                          '='
[45](1, 1, 46)      |                raw:                                          'foo'
[48](1, 1, 49)      |                raw:                                          '.'
[49](1, 1, 50)      |                raw:                                          'bat'
[52](1, 1, 53)      |                whitespace:                                   ' '
[53](1, 1, 54)      |                raw:                                          'and'
[56](1, 1, 57)      |                whitespace:                                   ' '
[57](1, 1, 58)      |                raw:                                          'bat'
[60](1, 1, 61)      |                raw:                                          '.'
[61](1, 1, 62)      |                raw:                                          'baz'
[64](1, 1, 65)      |                whitespace:                                   ' '
[65](1, 1, 66)      |                raw:                                          '<'
[66](1, 1, 67)      |                whitespace:                                   ' '
[67](1, 1, 68)      |                raw:                                          'foo'
[70](1, 1, 71)      |                raw:                                          '.'
[71](1, 1, 72)      |                raw:                                          'baz'
[74](1, 1, 75)      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:  22 |  PRS | Found unparsable section: ' FROM bat WHERE bat.bar=foo.bat and
                       | bat....'

```

## Steps to Reproduce

Put that SQL in a file, lint it :-)

## Version
Include the output of `sqlfluff --version` along with your Python version

sqlfluff, version 0.4.1
Python 3.9.2

## Configuration

None

**Comment by barrywhart** [comment]
This is a duplicate of #855, which is fixed in `master` and will be in the next release. I tested the `UPDATE` statement above, and it parses correctly now.

```
➜  sqlfluff git:(master) ✗ sqlfluff parse test.sql
[0](1, 1, 1)        |file:
[0](1, 1, 1)        |    statement:
[0](1, 1, 1)        |        update_statement:
[0](1, 1, 1)        |            keyword:                                          'UPDATE'
[6](1, 1, 7)        |            whitespace:                                       ' '
[7](1, 1, 8)        |            table_reference:
[7](1, 1, 8)        |                identifier:                                   'foo'
[10](1, 1, 11)      |            whitespace:                                       ' '
[11](1, 1, 12)      |            set_clause_list:
[11](1, 1, 12)      |                keyword:                                      'SET'
[14](1, 1, 15)      |                [META] indent:
[14](1, 1, 15)      |                whitespace:                                   ' '
[15](1, 1, 16)      |                set_clause:
[15](1, 1, 16)      |                    column_reference:
[15](1, 1, 16)      |                        identifier:                           'quux'
[19](1, 1, 20)      |                    comparison_operator:                      '='
[20](1, 1, 21)      |                    literal:                                  '42'
[22](1, 1, 23)      |                [META] dedent:
[22](1, 1, 23)      |            whitespace:                                       ' '
[23](1, 1, 24)      |            from_clause:
[23](1, 1, 24)      |                keyword:                                      'FROM'
[27](1, 1, 28)      |                whitespace:                                   ' '
[28](1, 1, 29)      |                from_expression:
[28](1, 1, 29)      |                    [META] indent:
[28](1, 1, 29)      |                    from_expression_element:
[28](1, 1, 29)      |                        table_expression:
[28](1, 1, 29)      |                            table_reference:
[28](1, 1, 29)      |                                identifier:                   'bat'
[31](1, 1, 32)      |                    [META] dedent:
[31](1, 1, 32)      |            whitespace:                                       ' '
[32](1, 1, 33)      |            where_clause:
[32](1, 1, 33)      |                keyword:                                      'WHERE'
[37](1, 1, 38)      |                [META] indent:
[37](1, 1, 38)      |                whitespace:                                   ' '
[38](1, 1, 39)      |                expression:
[38](1, 1, 39)      |                    column_reference:
[38](1, 1, 39)      |                        identifier:                           'bat'
[41](1, 1, 42)      |                        dot:                                  '.'
[42](1, 1, 43)      |                        identifier:                           'bar'
[45](1, 1, 46)      |                    comparison_operator:                      '='
[46](1, 1, 47)      |                    column_reference:
[46](1, 1, 47)      |                        identifier:                           'foo'
[49](1, 1, 50)      |                        dot:                                  '.'
[50](1, 1, 51)      |                        identifier:                           'bar'
[53](1, 1, 54)      |                    whitespace:                               ' '
[54](1, 1, 55)      |                    binary_operator:                          'and'
[57](1, 1, 58)      |                    whitespace:                               ' '
[58](1, 1, 59)      |                    column_reference:
[58](1, 1, 59)      |                        identifier:                           'bat'
[61](1, 1, 62)      |                        dot:                                  '.'
[62](1, 1, 63)      |                        identifier:                           'baz'
[65](1, 1, 66)      |                    whitespace:                               ' '
[66](1, 1, 67)      |                    comparison_operator:                      '<'
[67](1, 1, 68)      |                    whitespace:                               ' '
[68](1, 1, 69)      |                    column_reference:
[68](1, 1, 69)      |                        identifier:                           'foo'
[71](1, 1, 72)      |                        dot:                                  '.'
[72](1, 1, 73)      |                        identifier:                           'baz'
[75](1, 1, 76)      |                [META] dedent:
[75](1, 1, 76)      |    newline:
```
